src/openforms/registrations/contrib/objects_api/tests/test_backend.py

```diff
@@ -220,3 +227,172 @@ def test_registration_works_after_failure(self, m: requests_mock.Mocker):
 
         self.assertEqual(registration_data.pdf_url, "https://example.com")
         self.assertEqual(registration_data.csv_url, "")
+
+
+class ObjectsAPIBackendVCRTests(OFVCRMixin, TestCase):
+    VCR_TEST_FILES = Path(__file__).parent / "files"
+
+    def setUp(self):
+        super().setUp()
+
+        config_patcher = patch(
+            "openforms.registrations.contrib.objects_api.models.ObjectsAPIConfig.get_solo",
+            return_value=ObjectsAPIConfig(),
+        )
+        self.mock_get_config = config_patcher.start()
+        self.addCleanup(config_patcher.stop)
+
+    def test_submission_with_objects_api_backend_create_and_update_object(self):
+        objects_api_group = ObjectsAPIGroupConfigFactory.create(
+            for_test_docker_compose=True
+        )
+        plugin = ObjectsAPIRegistration(PLUGIN_IDENTIFIER)
+        form = FormFactory.create(
+            generate_minimal_setup=True,
+            formstep__form_definition__slug="fd-test",
+            formstep__form_definition__configuration={
+                "components": [
+                    {
+                        "type": "textfield",
+                        "key": "voornaam",
+                        "label": "Voornaam",
+                    },
+                ]
+            },
+        )
+        form_step = form.formstep_set.get()
+
+        # We manually create the objects instance as if it was created upfront by some external party
+        with get_objects_client(objects_api_group) as client:
+            created_obj = client.create_object(
+                record_data=prepare_data_for_registration(
+                    data={
+                        "name": {
+                            "voornaam": "My last name",
+                        },
+                    },
+                    objecttype_version=3,
+                ),
+                objecttype_url="http://objecttypes-web:8000/api/v2/objecttypes/8e46e0a5-b1b4-449b-b9e9-fa3cea655f48",
+            )
+
+        with self.subTest("Create an object"):
+            submission_create = SubmissionFactory.create(form=form, completed=True)
+            SubmissionStepFactory.create(
+                submission=submission_create,
+                form_step=form_step,
+                data={"voornaam": "My last name"},
+            )
+            object_create_result = plugin.register_submission(
+                submission_create,
+                {
+                    "version": 1,
+                    "objecttype": UUID("8e46e0a5-b1b4-449b-b9e9-fa3cea655f48"),
+                    "objecttype_version": 3,
+                    "objects_api_group": objects_api_group,
+                    "update_existing_object": False,
+                },
+            )
+
+            assert object_create_result is not None
+
+            self.assertNotEqual(object_create_result["uuid"], created_obj)
```

Comment:
this compares a dict with a string, so I think it should be:

```suggestion
            self.assertNotEqual(object_create_result["uuid"], created_obj["uuid"])
```